AutoSteamApp/Automaton/SteamworkAutomaton.cs

```diff
@@ -0,0 +1,348 @@
+﻿using AutoSteamApp.Configuration;
+using AutoSteamApp.Helpers;
+using AutoSteamApp.ProcessMemory;
+using GregsStack.InputSimulatorStandard;
+using GregsStack.InputSimulatorStandard.Native;
+using Logging;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Windows.Forms;
+
+namespace AutoSteamApp.Automaton
+{
+
+    public class SteamworkAutomaton
+    {
+
+        #region Fields
+
+        /// <summary>
+        /// Process loaded in the constructor.
+        /// </summary>
+        Process _Process;
+
+        /// <summary>
+        /// Save data loaded in the constructor.
+        /// </summary>
+        SaveData _SaveData;
+
+        /// <summary>
+        /// Steamworks data loaded in constructor.
+        /// </summary>
+        SteamworksData _SteamworksData;
+
+        /// <summary>
+        /// Field used to flag whether or not the currently running MHW:IB version is supported.
+        /// </summary>
+        bool _SupportedVersion = false;
+
+        /// <summary>
+        /// The input simulator used to mock button presses.
+        /// </summary>
+        InputSimulator _InputSimulator;
+
+        #endregion
+
+        #region Constructor
+
+        /// <summary>
+        /// Constructor which loads all required process memory and configuration values.
+        /// </summary>
+        public SteamworkAutomaton()
+        {
+            try
+            {
+                LoadAndVerifyProcess();
+                if (_SupportedVersion && !ConfigurationReader.RandomRun)
+                {
+                    _SteamworksData = new SteamworksData(_Process);
+                    _SaveData = new SaveData(_Process);
+                }
+                _InputSimulator = new InputSimulator();
+            }
+            catch (Exception e)
+            {
+                Log.Exception(new Exception("Failed to initialze Steamworks Automaton\n\t", e));
+                Log.Warning("Something went wrong reading the MHW:IB process.");
+                if (!ConfigurationReader.ShouldAutoQuit)
+                {
+                    Log.Message("Press any key to exit.");
+                    Console.ReadKey();
+                }
+                Environment.Exit(-1);
+
```

Comment:
Please avoid unnecessary empty lines - lint error when adding an empty line before a }

Reply:
Are you ok to leave any non-glaring lint issues to the next pull? I'm hesitant to add a linter in this pr when it is already so big. (I'm also sorry this pr is so big).